test/cmocka/src/math/numbers/gcd.c

```diff
@@ -27,18 +27,87 @@ static void test_math_numbers_gcd_for_12_and_9_equals_3(void **state)
 {
 	int r;
 
-	(void) state;
+	(void)state;
 
 	r = gcd(12, 9);
 	assert_int_equal(r, 3);
 }
 
+static void test_math_numbers_gcd_for_5_and_0_equals_5(void **state)
+{
+	int r;
+
+	(void)state;
+
+	r = gcd(5, 0);
+	assert_int_equal(r, 5);
+}
+
+static void test_math_numbers_gcd_for_0_and_5_equals_5(void **state)
+{
+	int r;
+
+	(void)state;
+
+	r = gcd(0, 5);
+	assert_int_equal(r, 5);
+}
+
+static void test_math_numbers_gcd_for_0_and_0_equals_0(void **state)
+{
+	int r;
+
+	(void)state;
+
+	r = gcd(5, 0);
+	assert_int_equal(r, 5);
+}
```

Comment:
Test name and contents seem inconsistent? If the contents is correct this is a duplicate test. So I'll assume the name is correct?

Reply:
The fix for this comment doesn't need to be a separate commit. You're editing something introduced in the same pull request so they could be combined into the commit that introduces it.

Reply:
The test cases were introduced two commits down the line. @paulstelian97 are you suggesting to amend the last changes to the previous commit?

Reply:
Yes, you should amend the changes in the commit that introduces the test case itself. Fixup patches only come when you fix something that is already merged upstream (say, in the master branch in case of firmware).

When fixing stuff within a pull request that wasn't merged yet rebase is recommended so you edit the commit itself, whenever reasonable (I believe this is one of the reasonable cases)